### ISSUE TITLE
Add extension point to define new tabs in the main area

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#5335]
The Wikidata extension currently hacks its way into the core UI to define tabs for the Schema editor, Issues and Preview.
This is not a healthy practice: by having this extension do this, we are encouraging other extensions to do the same, and this is a problem for multiple reasons:
* any change in the DOM of the grid view can break those extensions
* two extensions which both do that are likely to be incompatible, since the tabs they define will not be handled by the same code.
* correctly handling resize events has led us to add a reference to `SchemaAlignment` in the main code, which should be avoided at all costs because this is a solution that other extensions (not shipped with OpenRefine) cannot rely on (#5334).

### Proposed solution

Introduce an extension point, which can be used to register new tabs for this panel. Tabs are instances of ES6 classes which have various methods:
* `renderBody`, to render the body of the tab in a `<div>` supplied as argument
* `renderHeader`, to render the title similarly
* `resize`, to react to a resize event
* probably other methods that will turn out to be needed?

### Alternatives considered

Originally the Wikidata extension did not define tabs, but used dialogs instead. That was much less user friendly, but in architectural terms it was much cleaner since it relied on the existing dialog system.

### Additional context

Prompted by #5334.

**Comment by Abbe98** [comment]
Given this, would we like the tab row / header to always look like a tab-row? It's a little odd right now when it changes between the both modes. We could make it look just like the sidebars tabs but with the default of just having one?

![Screenshot 2023-05-20 at 21-08-43 fornvannen articles csv - OpenRefine](https://github.com/OpenRefine/OpenRefine/assets/2631719/fdb8d367-82c9-4f38-be8a-7726065ec88e)


![Screenshot 2023-05-20 at 21-09-26 fornvannen articles csv - OpenRefine](https://github.com/OpenRefine/OpenRefine/assets/2631719/9ae0de80-0181-4c54-bf2f-74a207bc686e)

**Comment by thadguidry** [comment]
@Abbe98 What are sidebars tabs?  Oh do you mean the Facet panel tabs?

**Comment by Abbe98** [comment]
> @Abbe98 What are sidebars tabs? Oh do you mean the Facet panel tabs?

Yes exactly.

**Comment by Abbe98** [comment]
Pinging @OpenRefine/designers and @wetneb

**Comment by wetneb** [comment]
I would find it natural to have the grid wrapped in a tab by default, indeed.

**Comment by Abbe98** [comment]
"Mockups"

Without tabs:
![Screenshot 2023-06-21 at 17-25-25 SHM - Fältinsamling - OpenRefine](https://github.com/OpenRefine/OpenRefine/assets/2631719/1e9707ed-54e8-4d27-973c-5047c7875497)

With tabs:
![Screenshot 2023-06-21 at 17-25-04 SHM - Fältinsamling - OpenRefine](https://github.com/OpenRefine/OpenRefine/assets/2631719/09cb5efc-73e2-4336-bd15-3c7180c0fb96)

**Comment by thadguidry** [comment]
@Abbe98 I would make all the tabs grey color and also add bordering in dark grey to each. The reason is that the blue tab doesn't look like a tab but a link next to first tab

**Comment by tfmorris** [comment]
Having the bar always styled as tabs is an improvement, I think. The mockup follows the current visual style (black active, blue not active), so I wouldn't change this without a lot of consideration. If I had to guess, the original design concept might have been blue == clickable, whether it be a link or tab or setting (e.g. rows per page, record/row mode). Other conventions are certainly possible, but this one has been in place for over a decade.

While we're picking nits though, I think the sharp cornered tabs (in the left bar and proposed top bar) look odd when all the other corners in the UI our radiused. I'd prefer to see everything rounded consistently.

**Comment by Abbe98** [comment]
> @Abbe98 I would make all the tabs grey color and also add bordering in dark grey to each. The reason is that the blue tab doesn't look like a tab but a link next to first tab

@thadguidry I agree, but I think such an endeavor deserves some work from an actual designer. At the same time I don't want to expand the scope here to affect other tab-menus.

> While we're picking nits though, I think the sharp cornered tabs (in the left bar and proposed top bar) look odd when all the other corners in the UI our radiused. I'd prefer to see everything rounded consistently.

Yeah couldn't agree more.

**Comment by Abbe98** [comment]
> correctly handling resize events has led us to add a reference to SchemaAlignment in the main code, which should be avoided at all costs because this is a solution that other extensions (not shipped with OpenRefine) cannot rely on (https://github.com/OpenRefine/OpenRefine/pull/5334).

I went ahead and dropped the need for the Wikibase schema tab to deal with resizing in JavaScript. I would suggest that we do not create a `resize` callback but instead assume extension developers rely on CSS for managing layouts and responsiveness.
